### PR TITLE
Fix bug descriptors pmgdqh

### LINKED ISSUE #37: Possible bug on simultaneous transactions
vdms: f07a3ca95755532075317a225dd5b6fdc27e51ec

(gdb) bt
#0  0x00007ffff5fe5428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff5fe702a in __GI_abort () at abort.c:89
#2  0x00007ffff60277ea in __libc_message (do_abort=do_abort@entry=2,
   fmt=fmt@entry=0x7ffff6140ed8 "*** Error in `%s': %s: 0x%s ***\n")
   at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff603037a in malloc_printerr (ar_ptr=<optimized out>, ptr=<optimized out>,
   str=0x7ffff6140fa0 "double free or corruption (fasttop)", action=3) at malloc.c:5006
#4  _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3867
#5  0x00007ffff603453c in __GI___libc_free (mem=<optimized out>) at malloc.c:2968
#6  0x00007ffff7ba24d8 in PMGD::Transaction::commit() ()
  from /home/ragaad/vdms-dep/pmgd/lib/libpmgd.so
#7  0x00000000004b6f90 in VDMS::PMGDQueryHandler::process_query(PMGD::protobufs::Command const*, PMGD::protobufs::CommandResponse*) ()
#8  0x00000000004b743e in VDMS::PMGDQueryHandler::process_queries(std::vector<PMGD::protobufs::Command*, std::allocator<PMGD::protobufs::Command*> > const&, int, bool) ()
#9  0x00000000004ad720 in VDMS::PMGDQuery::run() ()
#10 0x00000000004ca2d7 in VDMS::DescriptorsCommand::get_set_path(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int&) ()
#11 0x00000000004cc802 in VDMS::AddDescriptor::construct_protobuf(VDMS::PMGDQuery&, Json::Value const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, Json::Value&) ()
#12 0x0000000000474880 in VDMS::QueryHandler::process_query(VDMS::protobufs::queryMessage&, VDMS::protobufs::queryMessage&) ()
#13 0x0000000000476d38 in VDMS::QueryHandler::process_connection(comm::Connection*) ()
#14 0x00000000004a9507 in VDMS::CommunicationManager::process_queue() ()
#15 0x00007ffff695857f in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#16 0x00007ffff6c2e6ba in start_thread (arg=0x7fff4bd2a700) at pthread_create.c:333
#17 0x00007ffff60b741d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109

### VCL Info
vcl: 24c8085
vcl: we made this change to make it compile with gcc6
ragaad@bdw1:~/vdms-dep/vcl$ git diff
diff --git a/src/TDBDenseDescriptorSet.cc b/src/TDBDenseDescriptorSet.cc
index fef444b..f9fa95f 100644
--- a/src/TDBDenseDescriptorSet.cc
+++ b/src/TDBDenseDescriptorSet.cc
@@ -35,6 +35,7 @@
#include <iomanip>
#include <cmath>
#include <cstring>
+#include <numeric>

#include "TDBDescriptorSet.h"

diff --git a/src/TDBSparseDescriptorSet.cc b/src/TDBSparseDescriptorSet.cc
index a66f947..08e0bdd 100644
--- a/src/TDBSparseDescriptorSet.cc
+++ b/src/TDBSparseDescriptorSet.cc
@@ -33,6 +33,7 @@
#include <iomanip>
#include <cmath>
#include <cstring>
+#include <numeric>

#include "TDBDescriptorSet.h"
#include <tiledb/tiledb.h>

### PMGD Info
pmgd: made this change to make it compile for gcc6
ragaad@bdw1:~/vdms-dep/pmgd$ git diff
diff --git a/include/iterator.h b/include/iterator.h
index dc64ade..a07fbc0 100644
--- a/include/iterator.h
+++ b/include/iterator.h
@@ -256,7 +256,7 @@ namespace PMGD {
    class PropertyList {
        static const unsigned chunk_size = 64;

-uint8_t _chunk0[];
+uint8_t _chunk0[0];

  class PropertySpace;
  bool find_property(StringID property, PropertyRef &p,

### PR DESCRIPTION
**Pull request by luisremis**
Fix #37 , #33.

### CODE REVIEW

File: tests/python/TestRetail.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
+#
+# The MIT License
```
Comment (vishakha041) [inline tests/python/TestRetail.py]: Could you please split out the test commit.

File: src/DescriptorsCommand.cc [hunk 2]
```diff
@@ -876,7 +877,7 @@ Json::Value FindDescriptor::construct_responses(
 
         if (cache.isMember("cache_obj_id")) {
             // TODO CHECK THIS UNSAFE ERASE
-            _cache_map.unsafe_erase(cache["cache_obj_id"].asInt64());
+            // _cache_map.unsafe_erase(cache["cache_obj_id"].asInt64());
```
Comment (vishakha041) [inline src/DescriptorsCommand.cc]: Any reason you have this line commented out? If not needed, just remove it

File: tests/python/TestEntities.py [hunk 3]
```diff
@@ -89,33 +94,69 @@ def findEntity(self, thID):
...
+    def test_runMultipleAdds(self):
+
+        # Test concurrent AddEntities
+        concurrency = 32
+        thread_arr = []
+        results = [None] * concurrency
+        for i in range(0,concurrency):
+            thread_add = Thread(target=self.addEntity,args=(i, results) )
+            thread_add.start()
+            thread_arr.append(thread_add)
```
Comment (vishakha041) [inline tests/python/TestEntities.py]: What are these set of changes for? I am lazy to figure it out myself :) seems like a different commit from a quick scan.

Comment (luisremis) [reply tests/python/TestEntities.py]: stuff for checking that concurrent read/writes worked. you only need to worry about it if the test fails :)

**Pull request merged**